sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1719,6 +1719,14 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val ARITHMETIC_OPERATION_OVERFLOW_CHECK = buildConf("spark.sql.arithmetic.checkOverflow")
+    .doc("If it is set to true (default), all arithmetic operations on non-decimal fields throw " +
```

Comment:
Should we default this to `true` in this PR? Or defer the change of default to a separate PR?

Deferring the change of default would be consistent with the approach taken in the DecimalType PR and we can always flip the default in a separate PR (at which point we can write a migration guide note, too).

Reply:
Yes, we can do that!